src/sinks/blackhole.rs

```diff
@@ -80,23 +81,19 @@ impl BlackholeSink {
 
 #[async_trait]
 impl StreamSink for BlackholeSink {
-    async fn run(&mut self, mut input: BoxStream<'_, Event>) -> Result<(), ()> {
-        while let Some(event) = input.next().await {
+    async fn run(&mut self, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        let mut chunks = input.chunks(128);
+        while let Some(events) = chunks.next().await {
             if let Some(rate) = self.config.rate {
                 let until = self.last.unwrap_or_else(Instant::now)
                     + Duration::from_secs_f32(1.0 / rate as f32);
```

Comment:
Should this still be `1 / rate` now that we are consuming chunks?

Reply:
Oh yeah, I meant to chase this with a PR removing this rate notion but I suppose the logic is bungled here. It needs to be multiplied by whatever the chunk size is.